In [152]:
import pandas as pd

rA= pd.read_csv('realAll.csv')

rA

날짜   월  사고수         날씨
0     1   1   53  눈\n박무\n연무
1     2   1   95         연무
2     3   1   79     박무\n연무
3     4   1   78         연무
4     5   1   68         연무
..   ..  ..  ...        ...
361  27  12   40      비\n연무
362  28  12   81  비\n박무\n연무
363  29  12   71  눈\n박무\n연무
364  30  12   89        NaN
365  31  12   98          눈

[366 rows x 4 columns]

In [150]:
rA[(rA['날씨'] == '눈\n박무\n연무')] = '1'

In [153]:
rA

날짜   월  사고수         날씨
0     1   1   53  눈\n박무\n연무
1     2   1   95         연무
2     3   1   79     박무\n연무
3     4   1   78         연무
4     5   1   68         연무
..   ..  ..  ...        ...
361  27  12   40      비\n연무
362  28  12   81  비\n박무\n연무
363  29  12   71  눈\n박무\n연무
364  30  12   89        NaN
365  31  12   98          눈

[366 rows x 4 columns]

In [16]:
# 흠.. 비만 오는 날로 하기에도 눈 오는날도 있고, 다른 변수도 좀 많은거 같기도하고..
# 흠.. 강수량으로 따진다면.. 눈을 측정 못함. 근데 날씨만 단순 비교하기엔 무리가 있던듯.

In [42]:
rA

날짜   월  사고수       날씨
0     1   1   53  눈 박무 연무
1     2   1   95       연무
2     3   1   79    박무 연무
3     4   1   78       연무
4     5   1   68       연무
..   ..  ..  ...      ...
361  27  12   40     비 연무
362  28  12   81  비 박무 연무
363  29  12   71  눈 박무 연무
364  30  12   89      NaN
365  31  12   98        눈

[366 rows x 4 columns]

In [154]:
rA = rA.replace('(.*)\n(.*)', r'\1 \2', regex=True)

In [155]:
print(rA.replace('(.*)\n(.*)', r'\1 \2', regex=True))

     날짜   월  사고수       날씨
0     1   1   53  눈 박무 연무
1     2   1   95       연무
2     3   1   79    박무 연무
3     4   1   78       연무
4     5   1   68       연무
..   ..  ..  ...      ...
361  27  12   40     비 연무
362  28  12   81  비 박무 연무
363  29  12   71  눈 박무 연무
364  30  12   89      NaN
365  31  12   98        눈

[366 rows x 4 columns]


In [156]:
rA.head(10)

날짜  월  사고수           날씨
0   1  1   53      눈 박무 연무
1   2  1   95           연무
2   3  1   79        박무 연무
3   4  1   78           연무
4   5  1   68           연무
5   6  1  113  비 눈 진눈깨비 박무
6   7  1  123         비 박무
7   8  1   93    비 진눈깨비 박무
8   9  1  103        박무 연무
9  10  1   90        박무 연무

In [157]:
rA = rA.replace('(.*)눈(.*)', r'\1 1\2', regex=True)

In [158]:
rA = rA.replace('(.*)비(.*)', r'\1 2\2', regex=True)

In [159]:
rA.head(10)

날짜  월  사고수             날씨
0   1  1   53        1 박무 연무
1   2  1   95             연무
2   3  1   79          박무 연무
3   4  1   78             연무
4   5  1   68             연무
5   6  1  113  비 눈 진 1깨 2 박무
6   7  1  123           2 박무
7   8  1   93    비 진 1깨 2 박무
8   9  1  103          박무 연무
9  10  1   90          박무 연무

In [160]:
rAnew = rA.replace('[^0-9]', '', regex=True)

In [161]:
rAnew.head(10)

날짜  월  사고수  날씨
0   1  1   53   1
1   2  1   95    
2   3  1   79    
3   4  1   78    
4   5  1   68    
5   6  1  113  12
6   7  1  123   2
7   8  1   93  12
8   9  1  103    
9  10  1   90

rAnew.to_csv('rAnew2.csv',index=False)

In [162]:
rAnew = rAnew.replace('(.*)12(.*)', r'\1 3\2', regex=True)

In [163]:
rAnew = rAnew.replace('(.*)21(.*)', r'\1 3\2', regex=True)

In [175]:
rAnew = rAnew.replace(' 3', '3', regex=True)

In [181]:
rAnew = rAnew.replace(' ', '', regex=True)

In [182]:
rAnew.head(10)

날짜  월  사고수 날씨
0   1  1   53  1
1   2  1   95   
2   3  1   79   
3   4  1   78   
4   5  1   68   
5   6  1  113  3
6   7  1  123  2
7   8  1   93  3
8   9  1  103   
9  10  1   90

In [89]:
rAnew.to_csv('rAnew2.csv',index=False)

pd.get_dummies(df['Duration'])

In [165]:
rAnew

날짜   월  사고수   날씨
0     1   1   53    1
1     2   1   95     
2     3   1   79     
3     4   1   78     
4     5   1   68     
..   ..  ..  ...  ...
361  27  12   40    2
362  28  12   81    2
363  29  12   71    1
364  30  12   89  NaN
365  31  12   98    1

[366 rows x 4 columns]

In [91]:
pd.get_dummies(rAnew['날씨'])

3  1  2
0    0   0  1  0
1    1   0  0  0
2    1   0  0  0
3    1   0  0  0
4    1   0  0  0
..  ..  .. .. ..
361  0   0  0  1
362  0   0  0  1
363  0   0  1  0
364  0   0  0  0
365  0   0  1  0

[366 rows x 4 columns]

In [166]:
snow = rAnew['날씨'] == '1'

a = rAnew[snow]


a

날짜   월  사고수 날씨
0     1   1   53  1
33    3   2  115  1
34    4   2  104  1
37    7   2  159  1
38    8   2  106  1
39    9   2   86  1
40   10   2  100  1
47   17   2  114  1
351  17  12  101  1
352  18  12  142  1
363  29  12   71  1
365  31  12   98  1

In [167]:
a.sum()

날짜              154
월                63
사고수            1249
날씨     111111111111
dtype: object

In [ ]:
# 12일, 사고수 1249번.

In [168]:
rain = rAnew['날씨'] == '2'

b = rAnew[rain]

b

날짜   월  사고수 날씨
6     7   1  123  2
26   27   1   66  2
27   28   1   78  2
30   31   1   93  2
42   12   2  100  2
..   ..  ..  ... ..
326  22  11   63  2
331  27  11  125  2
357  23  12  108  2
361  27  12   40  2
362  28  12   81  2

[127 rows x 4 columns]

In [169]:
b.sum()

날짜                                                  2029
월                                                    855
사고수                                                12180
날씨     2222222222222222222222222222222222222222222222...
dtype: object

In [170]:
# 127일 사고수 12180 번.

In [179]:
rsAll = rAnew['날씨'] == '3'

c = rAnew[rsAll]

c

날짜   월  사고수 날씨
5     6   1  113  3
7     8   1   93  3
18   19   1   82  3
46   16   2   98  3
52   22   2   90  3
63    4   3   66  3
112  22   4   93  3
344  10  12   85  3
347  13  12   45  3

In [180]:
c.sum()

날짜           120
월             38
사고수          765
날씨     333333333
dtype: object

In [109]:
# 9일 사고수 765번. 

In [110]:
# 비, 눈, 눈비 의 경우로 나누어봐도 사고수와는 크게 연관이 없다...

전체 = 35227

날씨 있는 날 사고수 = 24495
날씨 없는 날 사고수 = 10732

2020년은 366일. 
그중 날씨가 있는 날은 258일
없는 날은 108 일. 

사고는 날씨와 크게 상관이 없다..
오히려 날씨가 없는 날 사고가 1일에 평균 100건에 가깝고, (평균 99.37건)
날씨가 있는 날은 94.94 건으로 날씨가 없는 날보다 4건 넘게 적다.


눈   12일, 사고수 1249번.
비   127일 사고수 12180 번.
눈비 9일 사고수 765번. 
총 148일 사고수 14194번.

눈이 오는날은 평균 100건을 넘고, 비가 오는 날은 평균 95.90 
눈비가 둘다 오는날은 오히려 적은 85건으로. 

박무,연무 와 같이 구름낀 날은 110일 사고수는 10301건 평균 93.6으로


눈비[85건] < 구름낀날[93.6] < 날씨가 있는날 전체 (ex)눈,비,박무,연무)[94.94] 
              < 비가오는날[95.90] < 날씨없는날[99.37] < 눈오는날[104.08]

눈 오는 날은 확실히 사고가 많고, 그외 비나, 구름, 등의 날씨보다는 일반적인 날씨에서의 사고가 더 잦다.

In [184]:
rAnew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   날짜      366 non-null    int64 
 1   월       366 non-null    int64 
 2   사고수     366 non-null    int64 
 3   날씨      258 non-null    object
dtypes: int64(3), object(1)
memory usage: 11.6+ KB


In [183]:
rAnew['날씨'].astype(int)

ValueError: invalid literal for int() with base 10: ''